pizza_cutter_metadetect/masks.py

```diff
@@ -0,0 +1,450 @@
+import healsparse
+import numpy as np
+import tqdm
+import time
+
+from .gaia_stars import make_gaia_mask
+
+MASK_INTILE = 2**0
+MASK_NOSLICE = 2**1
+MASK_GAIA_STAR = 2**2
+MASK_SLICEDUPE = 2**3
+MASK_TILEDUPE = 2**4
+
+
+def _wrap_ra(ra):
+    msk = (ra < 0) & np.isfinite(ra)
+    while np.any(msk):
+        ra[msk] = ra[msk] + 360.0
+        msk = (ra < 0) & np.isfinite(ra)
+
+    msk = (ra > 360) & np.isfinite(ra)
+    while np.any(msk):
+        ra[msk] = ra[msk] - 360.0
+        msk = (ra > 360) & np.isfinite(ra)
+
+    return ra
+
+
+def in_unique_coadd_tile_region(
+    *, ra, dec, crossra0, udecmin, udecmax, uramin, uramax,
+):
+    """Compute a boolean mask of the same shape as the input ra, dec indicating
+    if the given position is in the unique region for a coadd tile.
+
+    Parameters
+    ----------
+    ra : array-like
+        The array of ra values for the objects.
+    dec : array-like
+        The array of dec values for the objects.
+    crossra0 : str
+        A string that is either 'Y' or 'N' indicating if the tile cross ra=0.
+    udecmin, udecmax, uramin, uramax : float
+        The min and max values for ra and dec indicating the unique coadd tile
+        region.
+
+    Returns
+    -------
+    flag : bool array
+        The boolean flag array with True indicating the (ra,dec) is in the
+        unique coadd tile region.
+    """
+    ra = ra.copy()
+
+    ra = _wrap_ra(ra)
+
+    if crossra0 == 'Y':
+        uramin = uramin - 360.0
+        msk = ra > 180.0
+        ra[msk] -= 360
+
+    in_coadd = (
+        (ra > uramin)
+        & (ra <= uramax)
+        & (dec > udecmin)
+        & (dec <= udecmax)
+    )
+    msk = np.isfinite(ra) & np.isfinite(dec)
+    in_coadd[~msk] = False
+
+    return in_coadd
+
+
+def truncate_to_unique_coadd_tile_region(
+    *, ra, dec, crossra0, udecmin, udecmax, uramin, uramax,
+):
+    """Truncate a set of ra,dec values to the unique tile region.
+
+    Parameters
+    ----------
+    ra : array-like
+        The array of ra values for the objects.
+    dec : array-like
+        The array of dec values for the objects.
+    crossra0 : str
+        A string that is either 'Y' or 'N' indicating if the tile cross ra=0.
+    udecmin, udecmax, uramin, uramax : float
+        The min and max values for ra and dec indicating the unique coadd tile
+        region.
+
+    Returns
+    -------
+    rat, dect : array-like
+        The truncated values.
+    """
+    rat = ra.copy()
+    rat = _wrap_ra(rat)
+    if crossra0 == 'Y':
+        uramin = uramin - 360.0
+        msk = rat > 180.0
+        rat[msk] -= 360
+    rat = np.clip(rat, uramin, uramax)
+    rat = _wrap_ra(rat)
+    dect = np.clip(dec, udecmin, udecmax)
```

Comment:
I might be missing something, but is this the operation you want?  This will move all the ra/dec values outside the unique box to the values at the edges.  So now you have an overabundance of points tracking the boundaries.  I would think that you want to remove the ones that are outside the box, not `np.clip` them.
Though reading below, maybe this is the intended operation.  In which case I think that the docstring needs a bit more explanation.

Reply:
Yes I need to add comments. I am using a grid of 250x250 boxes to build up the ra-dec range. These bound the full coadd tile. Thus I clip the ra-dec values to the unique ra-dec range.

Reply:
This bit of code is removed so resolving.